lib/ws.js

```diff
@@ -97,13 +193,33 @@ class BinanceWS {
     }
 
     onUserData(binanceRest, eventHandler, interval = 60000) {
-        return binanceRest.startUserDataStream()
-            .then((response) => {
-                setInterval(() => {
-                    binanceRest.keepAliveUserDataStream(response);
-                }, interval);
-                return this._setupWebSocket(eventHandler, response.listenKey);
-            });
+        const setupAuth = () => {
+            let intervalId;
+
+            return {
+                flushKeepAlive() {
+                    if (intervalId) {
+                        clearInterval(intervalId);
+                    }
+                },
+
+                getAuthenticatedPath() {
+                    return binanceRest.startUserDataStream()
+                        .then(response => {
+                            intervalId = setInterval(() => {
+                                binanceRest.keepAliveUserDataStream(response).catch(e => {
+                                    const msg = 'Failed requesting keepAliveUserDataStream for onUserData listener';
+                                    console.error(new Date(), msg, e);
```

Comment:
There are situations where the old "listen key" key is no longer recoverable/valid. Could be useful to clear this interval and perhaps also re-connect the user data stream (generating a new listen key), if no connection otherwise exists. 

Although if this "auto-reconnect with new key", it may be best for this to be optional (e.g a new property under `this._options` with a default value).

Reply:
Could also be a good time to address #70 here if you're willing.